core/src/utilities/geometry.cc

```diff
@@ -749,4 +749,41 @@ namespace jiminy
             }
         };
     }
+
+    HeightmapFunction stairs(
+        double stepWidth, double stepHeight, uint32_t stepNumber, double orientation)
+    {
+        const double interpDelta = 0.01;
+        const Eigen::Rotation2D<double> rot_mat(orientation);
+
+        return [stepWidth, stepHeight, stepNumber, rot_mat, interpDelta](
+                   const Eigen::Vector2d & pos, double & height, Eigen::Vector3d & normal) -> void
+        {
+            // Compute position in stairs reference frame
+            Eigen::Vector2d posRel = (rot_mat.inverse() * pos);
+            const double modPos = std::fmod(std::abs(posRel[0]), stepWidth * stepNumber * 2);
+
+            // Compute the default height and normal
+            uint32_t stairIndex = static_cast<uint32_t>(modPos / stepWidth);
+            int8_t staircaseSlopeSign = 1;
+            if (stairIndex >= stepNumber)
+            {
+                stairIndex = 2 * stepNumber - stairIndex;
+                staircaseSlopeSign = -1;
+            }
+            height = stairIndex * stepHeight;
+            normal = Eigen::Vector3d::UnitZ();
+
+            // Avoid unsupported vertical edge
+            const double posRelOnStep = std::fmod(modPos, stepWidth) / stepWidth;
+            if (1 - posRelOnStep < interpDelta)
+            {
+                height += staircaseSlopeSign * (stepHeight / interpDelta) *
+                          (posRelOnStep - (1.0 - interpDelta));
+                normal << -staircaseSlopeSign * stepHeight / interpDelta, 0.0, 1.0;
```

Comment:
Normalise by yourself at this point. It would be more efficient:
```cpp
const double slope = staircaseSlopeSign * stepHeight / interpDelta;
const double normInv = 1.0 / std::sqrt(1.0 + std::pow(slope, 2));
normal << -slope * normInv, 0.0, normInv;
```

Reply:
Ok, I hesitated for this one. Also, is there a max slope for the terrain or can it just be as high as desired as long as it is not inf ? I've inspired myself from `tile2dInterp1d` for the implementation but the slope depends on the step height.

Reply:
what do you mean ? it is exactly the same as you did yourself, just reformulated to avoid useless computations x)

Reply:
There is a "max slope". The point is making sure that the normal is not pointing upward instantly after crossing the step. So it depends in theory of the integration tilmestep and max possible velocity. But a rough guess can be estimated.

Reply:
But in `tiles` or in the new `stairs` it is, the wider the tile/stairs, the longer the interpolation length.

Reply:
Yes for `tiles` it is, as it should be. But the **minimum slope** should be absolute not relative. A relative maximum slope of `0.01` was enforced for simplicity but it is indeed "wrong". In principle, the minimum interpolation delta should be computing dynamically according to the size of tiles / stairs.

Reply:
feel free to fix this issue if you have time, it can indeed be consider a bug to some extent.

Reply:
Ok, but if the "max slope" is not a fixed constant but depends on the integration timestep and max possible velocity, I understand that it is not trivial.

Reply:
yes but a pessimistic fixed constant would be better a better guess than what we are currently doing I would say. Something like: `dt = 4ms`, `v = 10 m/s` -> `dx = 4cm`. So the minimum slope is 4%.

Reply:
We can keep it like this for now if it is not blocking you.

Reply:
It's not blocking me for now. I will reflect on it.